lib/UndoInstCombinePass.cpp

```diff
@@ -66,13 +67,35 @@ bool clspv::UndoInstCombinePass::runOnFunction(Function &F) {
   return changed;
 }
 
+VectorType *InferTypeForOpaqueLoad(LoadInst *load, uint64_t vec_size) {
+  assert(load);
+  assert(load->getPointerOperandTy()->isOpaquePointerTy());
+  // Calculate the largest vector that the target can be split into.
+  // If the indexes don't work at the resulting size then they won't work for a
+  // smaller size.
+  uint64_t new_size = clspv::SPIRVMaxVectorSize();
+  while (vec_size % new_size) { // will always break at new_size == 1
+    new_size -= 1;
+  }
+  uint64_t divisor = vec_size / new_size;
```

Comment:
That seems reasonable since they scalar types would have always started in the proposed set before the InstCombine pass.

Reply:
It sounds like you intended a change here that didn't materialize. Just want to circle back and see if you had forgot to push a change here?

Reply:
After your prompt I have reconsider this bit and modified it to consider the case of a 12 x i8 vector, since that would lead to a 4 x i24 previously and I believe that all openCL C integer values have a power of 2 bit width.
I also have reduced the search to either size 4 or 3, I don't believe there are any cases where a size 2 vector would work and a size 4 vector wouldn't. In either case the resulting vector's scalar type is now bigger than the final destination type, so a truncation is still required.